src/keas/kmi/testclient.py

```diff
@@ -15,33 +16,39 @@
 """
 __docformat__ = "reStructuredText"
 
+import os
 import sys
 import optparse
 import textwrap
-import urlparse
+try:
+    # Python 3
+    from urllib.parse import urlparse
+except ImportError:
+    # Python 2
+    from urlparse import urlparse
 
 from keas.kmi.facility import LocalKeyManagementFacility
 
 def ping(kmf):
-    pieces = urlparse.urlparse(kmf.url)
+    pieces = urlparse(kmf.url)
     conn = kmf.httpConnFactory(pieces.netloc)
     conn.request('GET', '/')
     response = conn.getresponse()
-    print response.status, response.reason
-    print
-    print response.read()
+    print(response.status, response.reason)
+    print()
+    print(response.read())
 
 
 def new_key(kmf):
-    sys.stdout.write(kmf.generate())
+    os.write(sys.stdout.fileno(), kmf.generate())
```

Comment:
An interesting approach!  I'd've gone with `getattr(sys.stdout, 'buffer', sys.stdout).write(kmf.generate())`.

I don't have any arguments in favour of one or the other approach.  sys.stdout.buffer is easier to stub in unit tests, but os.write(...fileno(), ...) is shorter.  It's fine.